### ISSUE TITLE
Outfitter UI improvements

### ISSUE THREAD

**Issue by CodeKatta** [endless-sky/endless-sky#8757]
### Problem Description

Show if an item fits on your ship (and if not, how much space is needed) next to the image/name in the item panel.
Additionally, if an item is selected show on which ships it is installed (and maybe how many times).
Example below↓

### Related Issue Links

Couldn't find one that's directly related but these three also address improvements/enhancements to the OutfitterUI:
#7888 
#8539 
#8735 

### Desired Solution

I made a quick draft about how I think a solution might look like:

![Endless Sky_Outfitter](https://github.com/endless-sky/endless-sky/assets/26285588/c5175675-7291-47d8-a099-d4b17fa7393a)

The selected 'Carrier' has 13 outfit slots free, so the green number shows how many times these items could be installed on your ship or if at all (red '0'). The yellow number shows how much space you have to free in order to install this object one more time than the top number allows. Of course, the limiting factor depends on the smallest value of [general outfit|weapon¹|engine¹]-space (¹where applicable).
The selected _Small Heat Shunt [7]_ can be installed one more (a fourth) time and if you want to have five on your ship you have to free an additional slot. For the _S-970 Regenerator [37]_ you have to chuck out something that occupies at least 24 slots (maybe your old shield generator).
I think such a feature would be quite useful especially if you have several objects with similar space requirements like engines (from different manufacturers) where you have to match the various types while using the engine-space as efficiently as possible.
Primarily, it is thought to be a quick visual cue whether an item fits on your ship ('green') or not ('red').

Also, if you have many ships you might forget which item is installed where (and how many times). In order to not have to sift through all your ships, it would be nice to have a quick indicator on which ones you can find said item, for instance if you want to replace it with a better alternative.
In the draft above all four 'Carriers' have three _Small Heat Shunts_, the fourth 'Rano'erek' has two, the second 'Cruiser' just one, two 'Shield Beetles' are equipped with four and the 'Dreadnought' has five or more (eh, not really, just for illustrative purposes; also IMO installing sth. >4✕ is quite rare).
With ammunition one could use some kind of progress bar to show how depleted your reserves are (0%→100%).

As an afterthought (maybe this would would need a new post) it would be nice to have a list with all the stored outfits you have on different planets. So far, on the map you only get told in a tooltip how many you have stored on which planet but you can't see what they are.
Maybe at the bottom of the outfitter-screen (below your licenses) a folder for each storage on an other planet (▷collapsed by default). Of course, the items there can not be traded or moved around, just displayed.
Or create a third tab in the info-screen (next to ships) with a complete list of your outfits and where they are (installed, cargo, storage on planet A, B, C…). At least with your ships you can see where they stick around, even if parked someplace else.

### Alternative Approaches

As described above.

**Comment by LazerLit** [comment]
I like this. At first I was thinking, _'why do we need it'_, but I have had situations where I have a bit more space left and just want to fit in something (anything) which might be useful and this would help a lot. If you could also filter out items which are too bulky (or perhaps sort by size?) that could be useful too.

On your afterthought; you can already get this information on the map screen. I just discovered this myself when looking at it now, but if you click on the system where you stored things there is an extra line for each item saying how many you have stored (see my screenshot).
![Screenshot 2023-05-18 at 13 32 43](https://github.com/endless-sky/endless-sky/assets/105928379/d3d47123-4546-4ca8-8ab2-7b8d34bbef35)

**Comment by CodeKatta** [comment]
> I like this. At first I was thinking, _'why do we need it'_, but I have had situations where I have a bit more space left and just want to fit in something (anything) which might be useful and this would help a lot.

Exactly, initially that would have been my reaction, too. However, after I had amassed a veritable collection of 'useful things™' and concentrated them all on one planet (especially at least one of every type of engine I could get hold of [looting or buying]),  I was able to play around a bit with different configurations trying to use the available space as efficiently as possible. And if you don't know all the numbers by heart and can do all the arithmetics within a fraction of a second that can be quite a pain and very often they differ only by a very narrow margin. So at least displaying the needed space (and if it fits at all) in a way that you can see it at a glance, would be a great help.

> If you could also filter out items which are too bulky (or perhaps sort by size?) that could be useful too.

Yes, an option to sort items by different criteria (especially outfit space) would be quite nice.

> On your afterthought; you can already get this information on the map screen. I just discovered this myself when looking at it now, but if you click on the system where you stored things there is an extra line for each item saying how many you have stored (see my screenshot).

I also found that out just recently as well (after my post) but I still think it's a somewhat convoluted way to get an overview of your outfits. A solution as described in my last paragraph with all the outfits you own in one neat list (similar to the ship list) would be really convenient and probably not that complex to implement either, since something similar already exists.

The second topic I mentioned above (that with the green dots) could be really useful if one has a huge fleet (especially with various ship types and configurations, as do I). If you want to replace an item with a better one, those markers (or any similar solution) can help to quickly locate those ships which have that object installed.

-------------------
_Now shhh. Be vewy vewy quiet, I'm in the Gwaveyawd hunting wabbi'hets._
